memory/src/main/java/org/opensearch/ml/memory/index/InteractionsIndex.java

```diff
@@ -257,14 +305,59 @@ void innerGetInteractions(String conversationId, int from, int maxResults, Actio
             client
                 .admin()
                 .indices()
-                .refresh(Requests.refreshRequest(INTERACTIONS_INDEX_NAME), ActionListener.wrap(r -> { client.search(request, al); }, e -> {
+                .refresh(Requests.refreshRequest(indexName), ActionListener.wrap(r -> { client.search(request, al); }, e -> {
```

Comment:
Why refresh for each search request ? Seems more efficient for refreshing each write (create/update) request

Reply:
@HenryL27 , do you have a strong reason to refresh for requests that are not writing? Can we remove the refresh for Get/Search, etc?